### PR TITLE
Showing Warnings in Status Bar

### LINKED ISSUE #204: Warnings are reported as errors in the status bar
Viper successfully verifies the following program reporting a warning, but the Viper IDE reports a red message "Verifying prog.vpr failed after 0.7 seconds with 1 errors" in the status bar.

```
method test() {
    assert Set() == Set();
}
```

### PR DESCRIPTION
**Pull request by ArquintL**
Warnings have been correctly shown in the code but the status bar only reported the total of errors & warnings (i.e. the count of all diagnostics). This PR changes this behavior by displaying the number of errors (i.e. count of diagnostics with severity `ERROR`) and warnings (i.e. count of all non-error diagnostics) in the status bar.

Closes #204

### CODE REVIEW

File: client/src/VerificationController.ts [hunk 1]
```diff
@@ -801,14 +801,14 @@ export class VerificationController {
                                 State.statusBarItem.update("$(x) " + msg, Color.ERROR);
                                 break;
                             case Success.TypecheckingFailed:
-                                msg = `Type checking ${params.filename} failed after ${Helper.formatSeconds(params.time)} with ${params.nofErrors} error${params.nofErrors == 1 ? "s" : ""}`;
+                                msg = `Type checking ${params.filename} failed after ${Helper.formatSeconds(params.time)} with ${params.nofErrors} error${params.nofErrors == 1 ? "" : "s"} and ${params.nofWarnings} warning${params.nofWarnings == 1 ? "" : "s"}`;
```
Comment (fpoli) [inline client/src/VerificationController.ts:804]: I'm still puzzled by the "Type checking ... failed" part of the message. The following program *succeeds* in Silicon, even if there are warnings. Silicon's output doesn't mention any failure.
```
method test() {
    assert Set() == Set();
}
```

Output:
```
Silicon 1.1-SNAPSHOT
Type checker warning: Unconstrained type parameter, substituting default type Int. (prog.vpr@2.17--2.26)
Silicon finished verification successfully in 3.31s.
```

Maybe Silicon and the ViperServer don't agree regarding what's a typechecking failure? I'm not sure.

Comment (ArquintL) [reply client/src/VerificationController.ts:804]: Now I think I understand what you mean. This needs further investigation but I the following function seems to determine success/failure just based on the created diagnostics: https://github.com/viperproject/viper-ide/blob/4bb300b27c6fcc14530f30117fdfe0d65bd0dacf/server/src/VerificationTask.ts#L444

Comment (ArquintL) [reply client/src/VerificationController.ts]: It is weird that the IDE basically replicates the entire logic when something is treated as success or failure. However, warnings are now treated like Silicon treats them, i.e. verification succeeds when there are no errors (but possibly warnings).

**Review by fpoli — APPROVED** [review]
The changes look good to me. Has something been fixed in the ViperServer too, such that #204 returns a `Success.Success` instead of a `Success.VerificationFailed`?

**Pull request merged**